generators/server/templates/src/main/java/package/config/couchbase/CustomCouchbaseRepositoryQuery.java.ejs

```diff
@@ -0,0 +1,44 @@
+<%#
+ Copyright 2013-2021 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.config.couchbase;
+
+import org.springframework.data.couchbase.core.CouchbaseOperations;
+import org.springframework.data.couchbase.repository.query.CouchbaseQueryMethod;
+import org.springframework.data.couchbase.repository.query.CouchbaseRepositoryQuery;
+import org.springframework.data.repository.core.NamedQueries;
+import <%= packageName %>.config.couchbase.CustomN1qlRepositoryQueryExecutor;
+
+public class CustomCouchbaseRepositoryQuery extends CouchbaseRepositoryQuery {
```

Comment:
`CouchbaseRepositoryQuery` is deprecated. Can you please explain exactly what the problem is that forces you to implement all this custom classes ?

Reply:
This was already there on the branch that I started working from, so I'm not exactly sure. However, I didn't want to remove it so as to not break anything that depended on it. If you have any suggestions for removing or changing it, I can make those changes.

Reply:
After testing without it, it seems that there is some broken tests in `AccountResourceIT` due to this error:
```
java.lang.IllegalArgumentException: Authentication object cannot be null
```

I think we can find a more concise solution for that. But leave it for now. I'll try to look at it ASAP.